pkg/downloader/chart_downloader.go

```diff
@@ -318,6 +377,19 @@ func VerifyChart(path, keyring string) (*provenance.Verification, error) {
 	return sig.Verify(path, provfile)
 }
 
+// VerifyInMemoryChart takes key to a chart in the the InMemoryCache and a keyring,
+// and verifies the chart.
+//
+// It assumes that a chart data is accompanied by a embedded provenance file whose
+// name is the archive file name plus the ".prov" extension.
+func VerifyInMemoryChart(ref, keyring string, chartData, provData *bytes.Buffer) (*provenance.Verification, error) {
```

Comment:
Let's make this function private for now (lower case first letter).

Reply:
Done